### ISSUE TITLE
Is there a way to perform conditional accumulation in the table?

### ISSUE THREAD

**Issue by awayme** [teableio/teable#1166]
For example, I have two columns:
a     b  
0     1  
1     2  
0     4  
1     6  
1     8  

I want column C to show the cumulative sum of column B up to the current row, excluding rows where column A equals 1.

**Comment by caoxing9** [comment]
formula field support logical case. It work for this.
<img width="662" alt="image" src="https://github.com/user-attachments/assets/7af2f2e4-1a20-43c3-951b-ee62fdbad08b" />

**Comment by awayme** [comment]
Is this "for" logic operation a feature in the development version? I'm using the Docker version: 1.5.2 alpha build 170, and I don't have this "for" operator in my logic functions.
![image](https://github.com/user-attachments/assets/9b8c7b64-599c-4a42-9c54-f7d5ff8a055a)

**Comment by awayme** [comment]
![image](https://github.com/user-attachments/assets/087cce0f-8979-4e3e-9bd7-d57133075fde)

Thank you for the reply, but sorry, I still don't understand. 
I've made another practical table to illustrate the calculation I'm hoping for, with the last column being what I need. I want to cumulatively add the scores in the second column row by row, but if the third column is 'in progress', I don't add, and I keep the previous number.

Your suggestion was to use logical judgments to handle this, which I understand, but my understanding is that you were referring to an addition operation within one row, without providing a way to perform vertical operations across rows.

What I had in mind was there should be a filter and a variable representing all values in a certain column up to the current row, which would enable this conditional cumulative addition, like sum(filter(ColumnValuesUntilCurPosition(), condition)).

Perhaps I misunderstood your meaning?

**Comment by kpodp0ra** [comment]
> sum(filter(ColumnValuesUntilCurPosition(), condition))

I don't think it's even possible to implement order-depended variables, as order in Teable is tied to view. 
It would be terribly confusing due to view nature, as if different data appeared in view A and different data in view B.

I can work around this with "Previous" field:

![](https://github.com/user-attachments/assets/d7c8e01b-b278-4043-8dee-ac8863795e1e)

<details>
  <summary>Number roll-up field</summary>
  
![](https://github.com/user-attachments/assets/7aa2a393-3666-4090-bacf-a2dfccde1432)
</details>


<details>
  <summary>Previous field</summary>
  
![](https://github.com/user-attachments/assets/9c478cc8-4dfa-42f0-be22-7d8adf1d1132)
</details>

**Comment by caoxing9** [comment]
> ![image](https://private-user-images.githubusercontent.com/2276104/397657712-087cce0f-8979-4e3e-9bd7-d57133075fde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2OTE4MDgsIm5iZiI6MTczNDY5MTUwOCwicGF0aCI6Ii8yMjc2MTA0LzM5NzY1NzcxMi0wODdjY2UwZi04OTc5LTRlM2UtOWJkNy1kNTcxMzMwNzVmZGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjBUMTA0NTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODg0NTkzY2JjMWJhMjY2OTY1MDA2ZTYwOGJhYWNhOGI2ZDUyYjIyODljYzI1MjlhMjVjMjFjYmQ1NTNmYmQ0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.YHEbRk_qeBrLubJycpIgRVRMy_yCHRCSIyZUK5M7XVk)
> 
> Thank you for the reply, but sorry, I still don't understand. I've made another practical table to illustrate the calculation I'm hoping for, with the last column being what I need. I want to cumulatively add the scores in the second column row by row, but if the third column is 'in progress', I don't add, and I keep the previous number.
> 
> Your suggestion was to use logical judgments to handle this, which I understand, but my understanding is that you were referring to an addition operation within one row, without providing a way to perform vertical operations across rows.
> 
> What I had in mind was there should be a filter and a variable representing all values in a certain column up to the current row, which would enable this conditional cumulative addition, like sum(filter(ColumnValuesUntilCurPosition(), condition)).
> 
> Perhaps I misunderstood your meaning?

I got you. Frankly, formula does not support such case.
Perhaps the comming feature which support script action in automation, you can write a custom logic to calculate and write the specific cell.

**Comment by awayme** [comment]
> > ![image](https://private-user-images.githubusercontent.com/2276104/397657712-087cce0f-8979-4e3e-9bd7-d57133075fde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2OTE4MDgsIm5iZiI6MTczNDY5MTUwOCwicGF0aCI6Ii8yMjc2MTA0LzM5NzY1NzcxMi0wODdjY2UwZi04OTc5LTRlM2UtOWJkNy1kNTcxMzMwNzVmZGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjBUMTA0NTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODg0NTkzY2JjMWJhMjY2OTY1MDA2ZTYwOGJhYWNhOGI2ZDUyYjIyODljYzI1MjlhMjVjMjFjYmQ1NTNmYmQ0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.YHEbRk_qeBrLubJycpIgRVRMy_yCHRCSIyZUK5M7XVk)
> > Thank you for the reply, but sorry, I still don't understand. I've made another practical table to illustrate the calculation I'm hoping for, with the last column being what I need. I want to cumulatively add the scores in the second column row by row, but if the third column is 'in progress', I don't add, and I keep the previous number.
> > Your suggestion was to use logical judgments to handle this, which I understand, but my understanding is that you were referring to an addition operation within one row, without providing a way to perform vertical operations across rows.
> > What I had in mind was there should be a filter and a variable representing all values in a certain column up to the current row, which would enable this conditional cumulative addition, like sum(filter(ColumnValuesUntilCurPosition(), condition)).
> > Perhaps I misunderstood your meaning?
> 
> I got you. Frankly, formula does not support such case. Perhaps the comming feature which support script action in automation, you can write a custom logic to calculate and write the specific cell.

Thanks! I'm really looking forward to seeing this feature implemented and released soon.

**Comment by awayme** [comment]
> tied to view

Thank you for your workaround; it's a viable solution for now. I had tried using Rollup before but failed because I was unaware that the "link to another table" field could link back to itself.

As you mentioned, indeed, the order is tied to the view, which poses a problem but also serves as a feature. This way, I can achieve different accumulative sums with different orders in various views, which is quite common in many business data analysis scenarios. I've used this feature in other similar products like Lurke.

Thank you once again for your workaround method.
